File: mtool/reference/legacy-mtool-templates/dbclasses_template_system_default/objectivecm/template-dataclass-list-initialize-by-deserialize.m
- (id) initWithDic:(NSDictionary*)dic
{
    self = [super init];
    if (self != nil) {
__ARRAY_INITIALIZE_BASE__
        if ([dic isKindOfClass:[NSArray class]]) {
            NSArray* array = (NSArray*)dic;
            for (id object in array) {
                [self addObject:[[__CLASS_NAME__ alloc] initWithDic:object]];
            }
        }
        // == START OF EDITABLE AREA FOR INIT WITH DIC ==
        // == END OF EDITABLE AREA FOR INIT WITH DIC ==
    }
    return self;
}

File: mtool/reference/legacy-mtool-templates/dbclasses_template_system_default/swift/template-dataclass-list-initialize-by-deserialize.swift
    init(json:Any?)
    {
__ARRAY_INITIALIZE_BASE__
        var array = json as? Array<Any>;
        if (array != nil) {
            for i in 0 ..< array!.count {
                let object = array![i];
                _backendArray.append(__ARRAY_ITEM_CLASS_(json:object));
            }
        }
        // == START OF EDITABLE AREA FOR INIT WITH DIC ==
        // == END OF EDITABLE AREA FOR INIT WITH DIC ==
    }
